bindings/python/include/aligator/python/visitors.hpp

```diff
@@ -17,10 +17,12 @@ struct ClonePythonVisitor : bp::def_visitor<ClonePythonVisitor<T>> {
   }
 };
 
-template <typename T>
-struct CreateDataPythonVisitor : bp::def_visitor<CreateDataPythonVisitor<T>> {
+template <typename T, typename DataType>
+struct CreateDataPythonVisitor
+    : bp::def_visitor<CreateDataPythonVisitor<T, DataType>> {
   template <typename Pyclass> void visit(Pyclass &obj) const {
-    obj.def("createData", &T::createData, bp::args("self"),
+    std::shared_ptr<DataType> (T::*createData)() const = &T::createData;
```

Comment:
To get the return type, another possibility would be to use one of the modern C++ type traits, as in this SO question:

https://stackoverflow.com/a/73422762

Or maybe a declval-decltype construct,

```cpp
using ReturnType = decltype(std::declval<T>().createData());
```

Would this work maybe ?